go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -2114,3 +2116,8 @@ func skipQueryPlanCache(options *querypb.ExecuteOptions) bool {
 func (tsv *TabletServer) getShard() string {
 	return tsv.sm.Target().Shard
 }
+
+// nilSession is a helper that returns an empty session.
+func nilSession() *executorcontext.SafeSession {
+	return executorcontext.NewSafeSession(nil)
+}
```

Comment:
Why do we need to use these nil SafeSessions everywhere instead of allowing a nil *executorcontext.SafeSession everywhere and adding nil checks (which we still need)?